src/google/protobuf/compiler/php/php_generator.cc

```diff
@@ -456,31 +456,50 @@ std::string PhpGetterTypeName(const FieldDescriptor* field,
   if (field->is_map()) {
     return "\\Google\\Protobuf\\Internal\\MapField";
   }
-  if (field->is_repeated()) {
-    return "\\Google\\Protobuf\\Internal\\RepeatedField";
-  }
+  std::string type;
   switch (field->type()) {
     case FieldDescriptor::TYPE_INT32:
     case FieldDescriptor::TYPE_UINT32:
     case FieldDescriptor::TYPE_SINT32:
     case FieldDescriptor::TYPE_FIXED32:
     case FieldDescriptor::TYPE_SFIXED32:
-    case FieldDescriptor::TYPE_ENUM: return "int";
+    case FieldDescriptor::TYPE_ENUM:
+      type = "int";
+      break;
     case FieldDescriptor::TYPE_INT64:
     case FieldDescriptor::TYPE_UINT64:
     case FieldDescriptor::TYPE_SINT64:
     case FieldDescriptor::TYPE_FIXED64:
-    case FieldDescriptor::TYPE_SFIXED64: return "int|string";
+    case FieldDescriptor::TYPE_SFIXED64:
+      type = "int|string";
+      break;
     case FieldDescriptor::TYPE_DOUBLE:
-    case FieldDescriptor::TYPE_FLOAT: return "float";
-    case FieldDescriptor::TYPE_BOOL: return "bool";
+    case FieldDescriptor::TYPE_FLOAT:
+      type = "float";
+      break;
+    case FieldDescriptor::TYPE_BOOL:
+      type = "bool";
+      break;
     case FieldDescriptor::TYPE_STRING:
-    case FieldDescriptor::TYPE_BYTES: return "string";
+    case FieldDescriptor::TYPE_BYTES:
+      type = "string";
+      break;
     case FieldDescriptor::TYPE_MESSAGE:
-      return absl::StrCat("\\", FullClassName(field->message_type(), options));
+      type = absl::StrCat("\\", FullClassName(field->message_type(), options));
+      break;
     case FieldDescriptor::TYPE_GROUP: return "null";
     default: assert(false); return "";
   }
+  if (field->is_repeated()) {
+    // accommodate for edge case with multiple types.
+    size_t start_pos = type.find('|');
+    if (start_pos != std::string::npos) {
+      type.replace(start_pos, 1, ">|\\Google\\Protobuf\\Internal\\RepeatedField<");
```

Comment:
Can we use `ArrayAccess` instead? `RepeatedField` is an internal type and we do not want to expose it to public interfaces.

Reply:
This makes sense to me as long as there are no methods of `RepeatedField` that we want IDEs to typehint (e.g., we want the users to _only_ see the objects as an array, and NOT as a `RepeatedField`).

Also, if we do that, we also do that for the [setter typehints](https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/compiler/php/php_generator.cc#L442), for consistency. In this case, I think the best approach would be to use `[]`, such as `@return string[]`.

As changes like this will effect most of the files in our repo, making them at the same time would be ideal. If that sounds good to you, I'll update this PR to remove `RepeatedField` entirely, in favor of the `type[]` syntax!

Reply:
That sounds good to me. The only caveat is that I'm not sure if we support assignment of PHP arrays to repeated fields or not.

Reply:
Ok sounds good, yes let's remove RepeatedField in getters and setters. If we use `type[]` we should be satisfying that by implementing `ArrayAccess`, yes?

Reply:
@haberman yes that's correct! `ArrayAccess` will handle anything annotated with `type[]`